packages/awslint/lib/rules/api.ts

```diff
@@ -126,7 +126,7 @@ apiLinter.add({
           return;
         }
 
-        if (type.type.fqn === e.ctx.core.constructClass.fqn) {
+        if (type.type.fqn === e.ctx.core.baseConstructClassFqn) {
```

Comment:
By passes loading of the type just to compare the fqn.
I am making the assumption here that our core types are well tested and definitely exist in the assembly. If not jsii would blow up anyway.